src/raftstore/store/peer.rs

```diff
@@ -992,6 +992,44 @@ impl Peer {
         }
     }
 
+    /// propose a snapshot request. Note that the `None` response means
+    /// it requires the peer to perform a read-index.
+    pub fn propose_snapshot(&mut self,
+                            req: RaftCmdRequest,
+                            metrics: &mut RaftProposeMetrics)
+                            -> Option<RaftCmdResponse> {
+        if self.pending_remove {
+            return None;
+        }
+        metrics.all += 1;
+
+        // TODO: deny no snapshot request.
```

Comment:
This method is made for snapshot requests, and it should deny none snapshot requests by returning an error. Later, I will introduce new errors for those situations, and replace [`Option`](https://github.com/pingcap/tikv/pull/2107/files#diff-00fecd746483bbccacdd9e9eb7f0a1acR47) with `Result`. So let's keep it as a reminder.